File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.py
def solution(numbers):
    answer = [-1 for _ in numbers]

    stack = []
    for i in range(len(numbers)-1):
        stack.append(i)
        if numbers[i] < numbers[i+1]:
            while stack and numbers[stack[-1]] < numbers[i+1]:
                answer[stack.pop()] = numbers[i+1]

    return answer

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    var answer = new Array(numbers.length).fill(-1);
    var stack = [];
    for(var i =0;i<numbers.length;i++){
        while(stack&&numbers[stack.at(-1)]<numbers[i]){
            answer[stack.pop()] = numbers[i];
        }
        stack.push(i);
    }
    return answer;
}